services/src/main/java/org/keycloak/protocol/oidc/DefaultTokenExchangeProvider.java

```diff
@@ -630,7 +630,7 @@ protected UserModel importUserFromExternalIdentity(BrokeredIdentityContext conte
             }
 
             if (context.getIdpConfig().isTransientUsers()) {
-                user = new LightweightUserAdapter(session, context.getAuthenticationSession().getParentSession().getId());
+                user = new LightweightUserAdapter(session, realm);
```

Comment:
```suggestion
                user = new LightweightUserAdapter(session, realm, context.getAuthenticationSession().getParentSession().getId());
```

Is it possible to use that one above instead of generated ID? Or any particular reason why ID needs to be changed to be generated?

If you change this way, you can also remove constructor with signature `public LightweightUserAdapter(KeycloakSession session, RealmModel realm)` as it would not used.

Reply:
As mentioned in #28241, the NullPointerException is caused by context.getAuthenticationSession() == null, for this reason, I removed that part. 
If you believe this removal is incorrect, I can investigate further to understand why context.getAuthenticationSession() of BrokeredIdentityContext is null.

In my tests, both context.getAuthenticationSession() and context.getBrokerSessionId() (of BrokeredIdentityContext) consistently return null.

Reply:
Ok, I applied your suggestions in the second commit. 
It is slightly different, but it ensures the NullPointerException is avoided in cases where the parent session is empty.